### PR TITLE
Add mindet to avoid div0 errors in spi fitter and fix alpha and I0 variance estimates

### PR DESCRIPTION
**Pull request by landmanbester**
Small fix to avoid divide by zero errors during spi fitting. Let's hold off merging for now, want to test this thoroughly first

### CODE REVIEW

**Comment by landmanbester** [comment]
@sjperkins I think this is good to go when you have a sec

**Comment by landmanbester** [comment]
Oh wait, need the HISTORY.rst entry

**Comment by landmanbester** [comment]
Just to document the changes, this PR fixes the variance estimates of alpha and I0 when the per band weights in the fit are only correct up to some scaling factor and adds a minimum cutoff to the determinant computation that can cause the GN optimisation to fail if starting too far away from a local minimum

File: africanus/model/spi/component_spi.py
Comment (sjperkins) [inline africanus/model/spi/component_spi.py:35]: Can this comment be removed?

**Review by sjperkins — APPROVED** [review]
I'm happy, I'll leave it up to you to decide whether you want to deal with comment nit.

File: africanus/model/spi/component_spi.py
Comment (landmanbester) [reply africanus/model/spi/component_spi.py]: It has been removed

**Pull request merged**
